lib/oai/provider/model/activerecord_caching_wrapper.rb

```diff
@@ -83,7 +83,7 @@ def next_set(token_string)
       if token.last * @limit + @limit < total
         select_partial(token)
       else
-        select_partial(token).records
+        select_partial(token).tap { |list| list.instance_variable_set(:@token, list.token.next(nil)) }
```

Comment:
instance_variable_set is really unfortunate, and indication of failure of API.  

I am not really following what's going on in this PR in general, I have to admit -- why is what was before an _access_ (asking `.records`) being replaced with a *write*? 

Is there any way to provide clear API so you don't need to `instance_variable_set`, as well as it being more transparent what's going on here why, by using a clearly named method with documentation?

Reply:
Thanks for this feedback - I've refactored to remove the `instance_variable_set ` and (hopefully) clarify things.